### DISCUSSION TITLE
Lesson 6: Error:  (code: 3, message: execution reverted, data: Some(String("0x255c9bfe")))

### DISCUSSION THREAD

**Discussion by phvaz** [Cyfrin/foundry-full-course-cu#958] [Q&A]
So, despite managing to deploy my contract I'm having a hard time interacting with it using  `cast send` and am receiving the following error:

`Error:  (code: 3, message: execution reverted, data: Some(String("0x255c9bfe")))`

Now, here's my deploy contract:

![Captura de tela 2023-10-03 111011](https://github.com/Cyfrin/foundry-full-course-f23/assets/116604133/9fce0a19-cc62-4214-a4c1-382e1591472c)

My 6-Lesson.sol file is the same as Patrick's

Also, I already tried to use anvil's RPC istead of the Sepolia one but I got the error `(code: -32000, message: intrinsic gas too high -- CallGasCostMoreThanGasLimit, data: None)` and  saw in some discussion here that the guy solved the problem by changing to a Sepolia RPC but is not working here.

**Comment by theirrationalone** [comment]
The Error you're getting isn't the barrier.....
woooohh, I scare if i spoiled the answer 😅. Maybe @usmanfarooq91 ! could help you.

Heyy @usmanfarooq91 , Sorry to disturb you buddy 😅, would you like to help @phvaz !

**Comment by 0xusmanf** [comment]
@theirrationalone Thanks for the call, @phvaz Can you please specify the full cast command you are using to interact with your contract and your contract code?

**Reply by phvaz** [reply]
Alright, here's the contract(it's the same as in Patrick's distribution code in Lesson 6): 

          // SPDX-License-Identifier: MIT
          
          pragma solidity ^0.8.18;
          
          import "./AFoundryCourseChallenge.sol";
          
          contract LessonSix is AFoundryCourseChallenge {
              string private constant LESSON_IMAGE = "ipfs://QmZ8oE5hYAsoBL2f4Lih7ZPsBEYEsKo6BdY8ZPsjNsXJ7M"; // TODO
              mapping(address user => string iceCreamOrFood) private s_addressToIceCreamOrFood;
          
              constructor(address fcn) AFoundryCourseChallenge(fcn) {}
          
              /*
               * CALL THIS FUNCTION!
               * 
               * I challenge you to try to call this with the cast command! 
               * 
               * @param yourFavoriteIceCream - Your favorite ice cream 🍦 (or food if ice cream isn't your thing). 
               * @param yourTwitterHandle - Your twitter handle. Can be a blank string.
               */
              function solveChallenge(string memory yourFavoriteIceCream, string memory yourTwitterHandle) external {
                  _updateAndRewardSolver(yourTwitterHandle);
                  s_addressToIceCreamOrFood[msg.sender] = yourFavoriteIceCream;
              }
          
              ////////////////////////////////////////////////////////////////////////////////////////////////////////////
              /////////////////// The following are functions needed for the NFT, feel free to ignore. ///////////////////
              ////////////////////////////////////////////////////////////////////////////////////////////////////////////
              function description() external pure override returns (string memory) {
                  return unicode"Cyfrin Foundry Full Course: ⚒️ 🌋 🦾 We are in foundry! And excited to eat ";
              }
          
              function extraDescription(address user) external view override returns (string memory) {
                  return s_addressToIceCreamOrFood[user];
              }
          
              function attribute() external pure override returns (string memory) {
                  return "Yoked, stoked, and ready to code alongside ripped jesus";
              }
          
              function specialImage() external pure override returns (string memory) {
                  return LESSON_IMAGE;
              }
          }

Now here's the cast command I'm utilizing:

`cast send 0x91033c72079F6E6b512b81A0c3E71B73D7fc1553 "solveChallenge(string, string)" "vanilla" "a" --rpc-url $SEPOLIA_RPC_URL --private-key $PRIVATE_KEY `

**Comment by 0xusmanf** [comment]
> (code: -32000, message: intrinsic gas too high -- CallGasCostMoreThanGasLimit, data: None)

You are getting this error because when you switched to anvil and you didn't switch your private key, you are supposed to use the private key from one of the anvil's accounts if you are interacting on anvil.

I deployed the same contract on sepolia [here](https://sepolia.etherscan.io/address/0xbb011378af1c1d424270ba78317a4894de2a23db) and used the same `cast send` command which you mentioned and it was successful.

> Error:  (code: 3, message: execution reverted, data: Some(String("0x255c9bfe")))

The reason you are getting this error could be that you have already minted the NFT using the account you are trying to send this `cast send` command from, try it with a different account and see if it works (the account should have test ETH to pay for the gas).

Or maybe there is something wrong with your deployed contract because I tried to interact from an account that hasn't minted the lesson 6 NFT before and got the same error.

There is nothing wrong with your `cast` command, it is your contract that is reverting.

**Comment by Randell68** [comment]
(Code=3) execution reverted is what i'm getting

**Reply by EngrPips** [reply]
Are you sure you are not minting with an account that has already minted?
